nilearn/_utils/cache.py

```diff
@@ -132,7 +131,7 @@ def cache(func, memory, ref_memory_level=2, memory_level=1, **kwargs):
         returned.
     """
 
-    if ref_memory_level <= memory_level or memory is None:
+    if ref_memory_level > memory_level or memory is None:
```

Comment:
As far as I can see, they're simply passed through (so could be user-specified), except here: https://github.com/nilearn/nilearn/blob/master/nilearn/input_data/base_masker.py#L170

Seems weird to set the `ref_memory_level` to `self.memory_level`, given the docstring--this would disable memory usage, in my understanding.  Which is admittedly very weak :)